### PR TITLE
Add endpoint to rename groups

### PR DESCRIPTION
**Pull request by CalumMcCall**
- PRO-1536
 - Also update docs with all group endpoints
 - Remove delete endpoint as this needs to be spec'd more in light of campaigns referencing groups now

### CODE REVIEW

File: docs/swagger/sota-device_registry.yml [hunk 1]
```diff
@@ -207,6 +217,46 @@ paths:
...
+      parameter:
+      - name: CreateGroupRequest
+        description: The group name and devices to compare
+        in: body
+        required: true
+        type:
+      - name: namespace
+        in: query
+        required: true
+        type: string
```
Comment (Danten) [inline docs/swagger/sota-device_registry.yml]: Same

File: device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala [hunk 2]
```diff
@@ -123,6 +123,10 @@ class Routes(namespaceExtractor: Directive1[Namespace],
         (get & path("group_info") & pathEnd) {
           listGroups(ns)
         } ~
+        (put & extractUuid & pathPrefix("group_info") & path("rename") & parameter('groupName.as[Name])) {
```
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: No, path implies that it is the final segment. pathPrefix states that there are further segments to be matched, in this case "rename".

File: docs/swagger/sota-device_registry.yml [hunk 3]
```diff
@@ -198,15 +200,42 @@ paths:
...
         type: string
       responses:
         200:
           description: OK
-          schema:
-            $ref: '#/definitions/GroupInfo'
+  /device_groups/from_attributes:
+    post:
+      description: Create a group from two devices' common attributes
+      parameter:
```
Comment (Danten) [inline docs/swagger/sota-device_registry.yml]: should be
parameters:

**Comment by simao** [comment]
But that was already the case right?

Anyway, if we want to do that in all routes, we can pass the directive to the resource and use that. ie
```
val parameterNamespace: Directive1[Namespace] = parameter('namespace.as[Namespace])
new Routes(parameterNamespace, messageBus).route ~
```

and `Routes` and other resources remain unchanged.

File: device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceRequests.scala [hunk 4]
```diff
@@ -93,7 +93,8 @@ trait DeviceRequests {
     Put(Resource.uri(api, id.underlying.get, "group_info")
       .withQuery(Query("namespace" -> namespace.get, "groupName" -> groupName.get)), json)
 
-  def deleteGroupInfo(id: Uuid, namespace: Namespace)
-                     (implicit ec: ExecutionContext): HttpRequest =
-    Delete(Resource.uri(api, id.underlying.get, "group_info").withQuery(Query("namespace" -> namespace.get)))
+  def renameGroup(id: Uuid, newGroupName: Name, namespace: Namespace)(implicit ec: ExecutionContext): HttpRequest = {
+    Put(Resource.uri(api, id.underlying.get, "group_info", "rename")
+      .withQuery(Query("namespace" -> namespace.get, "groupName" -> newGroupName.get)))
```
Comment (simao) [inline device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceRequests.scala]: See my comment also

File: docs/swagger/sota-device_registry.yml [hunk 5]
```diff
@@ -207,6 +217,46 @@ paths:
...
+        type: string
+      - name: groupName
+        description: The new name of the group
+        in: query
+        required: true
+        type: string
+      - name: namespace
+        in: query
+        required: true
+        type: string
```
Comment (CalumMcCall) [inline docs/swagger/sota-device_registry.yml]: I think I've addressed this now.

File: device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceRequests.scala [hunk 6]
```diff
@@ -39,10 +39,10 @@ trait DeviceRequests {
 
   def fetchDevice(uuid: Uuid)
                  (implicit s: Show[Uuid]): HttpRequest =
-    Get(Resource.uri(api, s.show(uuid)).withQuery(Query("namespace" -> "default")))
+    Get(Resource.uri(api, s.show(uuid)))
 
-  def searchDevice(namespace: Namespace, regex: String): HttpRequest =
-    Get(Resource.uri(api).withQuery(Query("namespace" -> namespace.get, "regex" -> regex)))
+  def searchDevice(regex: String): HttpRequest =
+    Get(Resource.uri(api).withQuery(Query("regex" -> regex)))
```
Comment (Danten) [inline device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceRequests.scala]: This one actually have namespace as a query parameter.

**Comment by CalumMcCall** [comment]
I think I've addressed everything now, can one of you take another look please?

**Comment by Danten** [comment]
Review Score: 11/11

**Pull request merged**